frontend/src/App.test.js

```diff
@@ -11,12 +12,28 @@ beforeAll(() => {
   global.Date.now = mockedDate.now;
 });
 
+jest.mock('axios');
+
 describe('App', () => {
   it('should be defined', () => {
     expect(App).toBeDefined();
   });
+
   it('should render correctly', () => {
     const tree = renderer.create(<App />).toJSON();
     expect(tree).toMatchSnapshot();
   });
+
+  it('should initiate GET request using axios() to servlet', () => {
+    const servletEndpoint = '/api/v1/search-match';
+    const mockMatchStatus = {matchStatus: 'true'};
+    const mockData = {data: mockMatchStatus};
+
+    axios.get.mockImplementation(() => Promise.resolve(mockData));
+    const responseData = axios.get(servletEndpoint);
+
+    responseData.then((response) => {
+      expect(response.data).toEqual(mockData);
+    }).catch((error) => {});
```

Comment:
IIRC one has to do something special to test promises in javascript. It's possible that this test will run to completion without ever running the `then` block. You can test this out by adding a false assertion into the body of the `then` block and seeing if the test still passes.

Reply:
I added the false assertion and saw that the test did run to completion without running the then block. I changed the test to use async await instead and also found that another way of doing this would involved adding done() to each of the then and catch blocks.